comp110-worksheet-6/OxoBoard.cs

```diff
@@ -11,37 +11,100 @@ public enum Mark { None, O, X };
 	public class OxoBoard
 	{
 		// Constructor. Perform any necessary data initialisation here.
+
+        Mark[,] GameBoard;
+
 		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+		public OxoBoard(int width = 3, int height = 3, int inARow = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            //Creating the board
+            GameBoard = new Mark[width, height];
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            return GameBoard[x, y];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            //Added validation for the users inputs
+            if (x < GameBoard.GetLongLength(0) && y < GameBoard.GetLongLength(1) && x > -1 && y > -1)
```

Comment:
`GetLength` would be fine rather than `GetLongLength` here -- unless you plan on supporting boards larger than 2^32 :)